numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -1481,6 +1481,49 @@ PyArray_EquivTypes(PyArray_Descr *type1, PyArray_Descr *type2)
 }
 
 
+/*
+ * This function returns true if a view can be safely created
+ * between the two types. This implies that PyArray_EquivTypes
+ * is true as well.
+ */
+NPY_NO_EXPORT unsigned char
+PyArray_ViewableTypes(PyArray_Descr *type1, PyArray_Descr *type2)
```

Comment:
Maybe we can encode the equiv in the name? Or pass in the cast safety? Passing it in would make it work in the ufunc code I think.

But if not its OK.